fsm/src/main/java/org/egov/fsm/plantmapping/validator/PlantMappingValidator.java

```diff
@@ -0,0 +1,103 @@
+package org.egov.fsm.plantmapping.validator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.validation.Valid;
+
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.fsm.plantmapping.config.PlantMappingConfiguration;
+import org.egov.fsm.plantmapping.service.PlantMappingUserService;
+import org.egov.fsm.plantmapping.util.PlantMappingConstants;
+import org.egov.fsm.plantmapping.web.model.PlantMapping;
+import org.egov.fsm.plantmapping.web.model.PlantMappingRequest;
+import org.egov.fsm.plantmapping.web.model.PlantMappingSearchCriteria;
+import org.egov.fsm.plantmapping.web.model.UserDetailResponse;
+import org.egov.fsm.util.FSMConstants;
+import org.egov.fsm.util.FSMErrorConstants;
+import org.egov.tracer.model.CustomException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class PlantMappingValidator {
+
+	@Autowired
+	private MDMSPlantMapValidator mdmsValidator;
+	
+	@Autowired
+	private PlantMappingUserService userService;
+	
+	@Autowired
+	private PlantMappingConfiguration config;
+	
+	public void validateCreateOrUpdate(@Valid PlantMappingRequest request, Object mdmsData) {
+		
+		if (StringUtils.isEmpty(request.getPlantMapping().getTenantId())) {
+			throw new CustomException(PlantMappingConstants.INVALID_TENANT, "TenantId is mandatory");
+		}
+		if (request.getPlantMapping().getTenantId().split("\\.").length == 1) {
+			throw new CustomException(PlantMappingConstants.INVALID_TENANT, "Invalid TenantId");
+		}
+		if (request.getPlantMapping().getEmployeeUuid() == null || request.getPlantMapping().getEmployeeUuid().isEmpty()) {
+			throw new CustomException(PlantMappingConstants.INVALID_UUID, "At lease one employee uuid is required");
+		}
+		if (request.getPlantMapping().getPlantCode() == null || request.getPlantMapping().getPlantCode().isEmpty()) {
+			throw new CustomException(PlantMappingConstants.INVALID_PLANT_CODE, "");
+		}
+		mdmsValidator.validateMdmsData(request, mdmsData);
+		
+		PlantMapping plantMap = request.getPlantMapping();
+		plantMap.getEmployeeUuid();
+		 if(!request.getRequestInfo().getUserInfo().getType().equalsIgnoreCase(FSMConstants.EMPLOYEE)) {
```

Comment:
plant can be mapped to the Employee with role FSTP Plant Operator only.

And need to verify if there is any existing plancode mapping for the same employee, if yes new mapping should not be allowed.